static/js/containers/ProductDetailEnrollApp.js

```diff
@@ -19,19 +20,130 @@ import {
+      addProductToBasket,
+      addUserNotification
+    } = this.props
+    const run = courseRuns ? courseRuns[0] : null
+    if (run === null) {
+      return
+    }
+    const product = run.products ? run.products[0] : null
+    if (product === null) {
+      return
+    }
+    const resp = await addProductToBasket(currentUser.id, product.id)
+    let userMessage, messageType
+    if (isSuccessResponse(resp)) {
+      messageType = ALERT_TYPE_SUCCESS
+      userMessage = "You have successfully added the course to your cart"
+    } else {
+      messageType = ALERT_TYPE_DANGER
+      userMessage = `Something went wrong trying to add course to cart. Please contact support at ${
+        SETTINGS.support_email
+      }.`
+    }
+    addUserNotification({
+      "add-product-status": {
+        type:  messageType,
+        props: {
+          text: userMessage
+        }
+      }
+    })
+    this.setState({ upgradeEnrollmentDialogVisibility: false })
+  }
+  toggleUpgradeDialogVisibility = () => {
+    const { upgradeEnrollmentDialogVisibility } = this.state
+    this.setState({
+      upgradeEnrollmentDialogVisibility: !upgradeEnrollmentDialogVisibility
+    })
+  }
+
+  renderUpgradeEnrollmentDialog() {
+    const { upgradeEnrollmentDialogVisibility } = this.state
+    return (
+      <Modal
+        id={`upgrade-enrollment-dialog`}
+        className="upgrade-enrollment-modal"
+        isOpen={upgradeEnrollmentDialogVisibility}
+        toggle={() => this.toggleUpgradeDialogVisibility()}
+      >
+        <ModalHeader toggle={() => this.toggleUpgradeDialogVisibility()}>
+          Enroll
+        </ModalHeader>
+        <ModalBody>
+          <div className="row modal-subheader">
+            <div className="col-10">Take a course and get a certificate</div>
+            <div className="col-2">$1000</div>
+          </div>
+          <div className="row">
+            <div className="col-4">
+              <div className="upgrade-icon" />
+            </div>
+            <div className="col-8">
+              <p>
+                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
+                eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut
+                enim ad minim veniam, quis nostrud exercitation ullamco laboris
+                nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor
+                in reprehenderit in voluptate velit esse cillum dolore eu fugiat
+                nulla pariatur. Excepteur sint occaecat cupidatat non proident,
+                sunt in culpa qui officia deserunt mollit anim id est laborum.
+              </p>
+              <button
+                onClick={this.addItemToBasket.bind(this)}
+                className="btn btn-primary btn-gradient-red"
+              >
+                Continue
+              </button>
+            </div>
+          </div>
+          <div className="cancel-link">
+            <a onClick={() => this.toggleUpgradeDialogVisibility()}>
```

Comment:
- When the user clicks this, it should also POST to `/enrollments/` like the legacy "Enroll" button did
- The link isn't getting the `pointer` cursor style on hover, this might be because it needs an `href` attribute, although that shouldn't go anywhere

Reply:
ohh, okay got it